src/core/connection.cpp

```diff
@@ -43,7 +43,7 @@ void Connection::stop_mavlink_receiver()
 
 void Connection::receive_message(mavlink_message_t& message)
 {
-    if (_system_ids.find(message.sysid) != _system_ids.end()) {
+    if (_system_ids.find(message.sysid) == _system_ids.end()) {
         _system_ids.insert(message.sysid);
```

Comment:
The problem was coming from here, I was not adding the new system_id when receiving a message from a new system_id.